api/app_analytics/serializers.py

```diff
@@ -12,6 +14,17 @@ class UsageDataSerializer(serializers.Serializer):
 class UsageDataQuerySerializer(serializers.Serializer):
     project_id = serializers.IntegerField(required=False)
     environment_id = serializers.IntegerField(required=False)
+    period = serializers.RegexField(
+        regex=r"^(%s)$"
+        % "|".join(
+            [
+                constants.CURRENT_BILLING_PERIOD,
+                constants.PREVIOUS_BILLING_PERIOD,
+                constants._90_DAY_PERIOD,
+            ]
+        ),
+        required=False,
+    )
```

Comment:
When I offered to use `RegexField` initially, I was thinking more of a `^\d+[d|h|m]$` regex, as I didn't know that the periods were meant to be predefined.

Now it seems more appropriate to use `ChoiceField` instead (and if you define the literal type outside the method definition as suggested [here](https://github.com/Flagsmith/flagsmith/pull/3870/files#r1604137042), you'll be able to use `typing.get_args` with it to provide the choices in a DRY manner).

Reply:
Yeah I should have gone with a `ChoiceField`

Reply:
Ok created the `ChoiceField` with the `typing.get_args` trick. Nice.